### PR TITLE
[MDS-5245] TS conversion incident

### PR DESCRIPTION
**Pull request by isuru-aot**
## Objective 

[MDS-5245](https://bcmines.atlassian.net/browse/MDS-5245)

Converted MineIncident in core-web to typescript.
Created an interface file
Converted MineIncident unit test file to typescript.
Convert related, Reducer and ActionCreator files.
Ran unit tests to ensure all tests are passing

### CODE REVIEW

File: services/common/src/interfaces/mineIncident.interface.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export interface IMineIncident {
+  mine_name: string;
+  categories: any;
+  incident_timestamp: any;
```
Comment (henryoforeh-dev) [inline services/common/src/interfaces/mineIncident.interface.ts]: Can you please change the data type here to string ?

File: services/core-web/src/components/mine/Incidents/MineIncident.tsx [hunk 2]
```diff
@@ -19,35 +19,46 @@ import { clearMineIncident } from "@common/actions/incidentActions";
 import * as Strings from "@common/constants/strings";
 import * as FORM from "@/constants/forms";
 import Loading from "@/components/common/Loading";
-import customPropTypes from "@/customPropTypes";
+// import customPropTypes from "@/customPropTypes";
```
Comment (henryoforeh-dev) [inline services/core-web/src/components/mine/Incidents/MineIncident.tsx]: Please remove this commented line of code.

File: services/core-web/src/components/mine/Incidents/MineIncident.tsx [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+import React, { FunctionComponent, Requireable, useEffect, useState } from "react";
 import { bindActionCreators } from "redux";
 import { flattenObject } from "@common/utils/helpers";
 import { connect } from "react-redux";
 import { Link, withRouter, useParams, useLocation, matchPath } from "react-router-dom";
 import { change, submit, getFormSyncErrors, getFormValues, touch, isDirty } from "redux-form";
 import { Tag } from "antd";
 import { ArrowLeftOutlined, EnvironmentOutlined } from "@ant-design/icons";
-import PropTypes from "prop-types";
+import PropTypes, { string } from "prop-types";
```
Comment (henryoforeh-dev) [inline services/core-web/src/components/mine/Incidents/MineIncident.tsx]: Please check if PropTypes is used anywhere here.

File: services/core-web/src/components/mine/Incidents/MineIncident.tsx [hunk 4]
```diff
@@ -19,35 +19,46 @@ import { clearMineIncident } from "@common/actions/incidentActions";
...
-  history: PropTypes.shape({ push: PropTypes.func, replace: PropTypes.func }).isRequired,
-  // eslint-disable-next-line react/no-unused-prop-types
-  formValues: PropTypes.objectOf(PropTypes.any).isRequired,
-  // eslint-disable-next-line react/no-unused-prop-types
-  formIsDirty: PropTypes.bool.isRequired,
-  formErrors: PropTypes.objectOf(PropTypes.string).isRequired,
-};
+export interface MineIncidentProps {
+  incident: IMineIncident;
+  createMineIncident(mineGuid: string, formattedValues: any): Promise<IMineIncident>;
```
Comment (matbusby-fw) [inline services/core-web/src/components/mine/Incidents/MineIncident.tsx]: There is now a generic type `ActionCreator` that can be used for any of the action creators in these interfaces.  It'll cut down on having to manually add in the props/returns here.

You can add this import:

import { ActionCreator } from "@/interfaces/actionCreator";

then change these to something like this:

`createMineIncident(mineGuid: ActionCreator<typeof createMineIncident>;`

File: services/core-web/src/components/mine/Incidents/MineIncident.tsx [hunk 5]
```diff
@@ -19,35 +19,46 @@ import { clearMineIncident } from "@common/actions/incidentActions";
...
-  const [isLoaded, setIsLoaded] = useState(false);
-  const [fixedTop, setIsFixedTop] = useState(false);
+interface IProps {
+  removeDocumentFromMineIncident: (IParams) => Promise<void>;
+  fetchMineIncident: (mineGuid: string, mineIncidentGuid: string) => Promise<void>;
+}
+
+export const MineIncident: FunctionComponent<MineIncidentProps> = (props) => {
+  const { formValues, formErrors, incident, history } = props;
+  const { mineGuid, mineIncidentGuid = null }: any = useParams();
```
Comment (matbusby-fw) [inline services/core-web/src/components/mine/Incidents/MineIncident.tsx]:
```suggestion
  const { mineGuid, mineIncidentGuid = null }: any = useParams<IParams>();
```

File: services/core-web/src/components/mine/Incidents/MineIncident.tsx [hunk 6]
```diff
@@ -216,40 +254,20 @@ export const MineIncident = (props) => {
...
-              { href: "documentation", title: "Documentation" },
-              { href: "final-report", title: "Final Report" },
-              { href: "ministry-follow-up", title: "Ministry Follow Up" },
-              { href: "internal-documents", title: "Internal Documents" },
-              { href: "internal-ministry-comments", title: "Comments" },
-            ]}
-            featureUrlRoute={sideBarRoute.url.hashRoute}
-            featureUrlRouteArguments={sideBarRoute.params}
-          />
+          <ScrollSideMenu {...scrollSideMenuProps} />
```
Comment (matbusby-fw) [inline services/core-web/src/components/mine/Incidents/MineIncident.tsx:255]: Tidy!  Love it!

**Review by matbusby-fw — CHANGES_REQUESTED** [review]
Looks awesome.  Just requested a couple small changes to use the new Generic `ActionCreator` type.  This will be more reliable in the future when changes are made to those action creators.

File: services/common/src/interfaces/mineIncident.interface.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+  mine_incident_guid: string;
```
Comment (taraepp) [inline services/common/src/interfaces/mineIncident.interface.ts:5]: You're probably going to have to make some adjustments with the timezone/timestamp now, once you pull those changes. :grin: 
The incident_timezone will be a string, and there are some values in strings.js now that should be the possible values if you wanted to go there (making a timezone type is probably scope creep). The timestamp will be an ISO string.

File: services/common/src/interfaces/mineIncident.interface.ts (hunk 1 above)
Comment (isuru-aot) [reply services/common/src/interfaces/mineIncident.interface.ts]: Updated to string.

File: services/core-web/src/components/mine/Incidents/MineIncident.tsx (hunk 2 above)
Comment (isuru-aot) [reply services/core-web/src/components/mine/Incidents/MineIncident.tsx]: Removed commented line and unused imports.

File: services/core-web/src/components/mine/Incidents/MineIncident.tsx (hunk 3 above)
Comment (isuru-aot) [reply services/core-web/src/components/mine/Incidents/MineIncident.tsx]: Removed unused imports

File: services/core-web/src/components/mine/Incidents/MineIncident.tsx (hunk 4 above)
Comment (isuru-aot) [reply services/core-web/src/components/mine/Incidents/MineIncident.tsx]: Updated with the generic type `ActionCreator`

File: services/core-web/src/components/mine/Incidents/MineIncident.tsx (hunk 5 above)
Comment (isuru-aot) [reply services/core-web/src/components/mine/Incidents/MineIncident.tsx]: Updated

File: services/common/src/interfaces/mineIncident.interface.ts (hunk 7 above)
Comment (isuru-aot) [reply services/common/src/interfaces/mineIncident.interface.ts:5]: Resolved the merge conflicts in `MineIncident.tsx` file.

**Review by asinn134 — APPROVED** [review]
nice work!

**Review by henryoforeh-dev — APPROVED** [review]
Nice job!

**Review by matbusby-fw — APPROVED** [review]
